app/quest-boost/claim/[boostId]/page.tsx

```diff
@@ -0,0 +1,207 @@
+import { TOKEN_ADDRESS_MAP } from "@utils/constants";
+
+type BoostQuestPageProps = {
+  params: {
+    boostId: string;
+  };
+};
+
+export default function Page({ params }: BoostQuestPageProps) {
+  const { address, account } = useAccount();
+  const { boostId } = params;
+  const [boost, setBoost] = useState<Boost>();
+  const [sign, setSign] = useState<Signature>(["", ""]);
+  const { addTransaction } = useNotificationManager();
+  const [displayCard, setDisplayCard] = useState<boolean>(false);
+  const [displayLottie, setDisplayLottie] = useState<boolean>(true);
+  const [transactionHash, setTransactionHash] = useState<string>("");
+
+  const fetchPageData = async () => {
+    const boostInfo = await getBoostById(boostId);
+    setBoost(boostInfo);
+  };
+
+  useEffect(() => {
+    fetchPageData();
+  }, []);
+
+  const fetchBoostClaimParams = async (): Promise<Signature> => {
+    let formattedSign: Signature = ["", ""];
+    try {
+      if (!boost?.id || !address) return formattedSign;
+      const res = await getQuestBoostClaimParams(boost.id);
+      formattedSign = [res?.r, res?.s];
+      return formattedSign;
+    } catch (err) {
+      console.log(err);
+      return formattedSign;
+    }
+  };
+
+  const handleClaimClick = async () => {
+    const signature = await fetchBoostClaimParams();
+    if (!signature) return;
+    setSign(signature);
+  };
+
+  const getTokenName = useCallback(() => {
+    if (!boost) return "";
+    const network = process.env.NEXT_PUBLIC_IS_TESTNET ? "TESTNET" : "MAINNET";
+    switch (boost.token) {
+      case TOKEN_ADDRESS_MAP[network].USDC:
+        return "USDC";
+      case TOKEN_ADDRESS_MAP[network].ETH:
+        return "ETH";
+      default:
+        return "USDC";
+    }
+  }, [boost]);
+
+  useEffect(() => {
+    if (!account || !boost || sign[0].length === 0 || sign[1].length === 0)
+      return;
+
+    const callContract = async () => {
+      const { transaction_hash } = await account.execute(
+        boostClaimCall(boost, sign)
+      );
+      BoostClaimStatusManager.updateBoostClaimStatus(boost?.id, true);
+      setTransactionHash(transaction_hash);
+    };
+
+    callContract();
+  }, [sign]);
+
+  useEffect(() => {
+    if (!(transactionHash.length > 0)) return;
+    if (transactionHash) {
+      addTransaction({
+        timestamp: Date.now(),
+        subtext: boost?.name ?? "Quest Boost Rewards",
+        type: NotificationType.TRANSACTION,
+        data: {
+          type: TransactionType.CLAIM_REWARDS,
+          hash: transactionHash,
+          status: "pending",
+        },
+      });
+    }
+  }, [transactionHash]);
+
+  return (
+    <div className={styles.claim_screen_container}>
+      <div className="flex flex-col gap-16">
+        {displayCard ? (
+          <>
+            <div className={styles.claim_amount_card}>
+              {boost &&
+              boost?.winner &&
+              hexToDecimal(boost?.winner) === hexToDecimal(address) ? (
```

Comment:
you're repeating this many times, maybe you could create a variable for it it, would improve readability

Reply:
done